saml-handler/pom.xml

```diff
@@ -0,0 +1,366 @@
+    this file except in compliance with the License. You may obtain a copy of
+    the License at http://www.apache.org/licenses/LICENSE-2.0 Unless required
+    by applicable law or agreed to in writing, software distributed under the
+    License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS
+    OF ANY KIND, either express or implied. See the License for the specific
+    language governing permissions and limitations under the License. -->
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+  <parent>
+    <groupId>org.apache.sling</groupId>
+    <artifactId>sling</artifactId>
+    <version>37</version>
+    <relativePath />
+  </parent>
+
+  <modelVersion>4.0.0</modelVersion>
+  <groupId>org.apache.sling.auth</groupId>
+  <artifactId>saml2</artifactId>
+  <version>0.1.0-SNAPSHOT</version>
+  <name>SAML2 Service Provider</name>
+
+  <properties>
+    <opensaml.version>3.4.3</opensaml.version>
+    <sling.host>localhost</sling.host>
+    <sling.port>8080</sling.port>
+    <sling.user>admin</sling.user>
+    <sling.password>admin</sling.password>
+  </properties>
+
+  <packaging>bundle</packaging>
+
+  <build>
+    <plugins>
+      <plugin>
+        <groupId>org.apache.felix</groupId>
+        <artifactId>maven-bundle-plugin</artifactId>
+        <version>3.5.0</version>
+        <extensions>true</extensions>
+        <configuration>
+          <instructions>
+            <Bundle-Activator>org.apache.sling.auth.saml2.Activator</Bundle-Activator>
+            <Embed-Dependency>*;scope=compile|runtime;</Embed-Dependency>
+            <Embed-Transitive>true</Embed-Transitive>
+            <Embed-Directory>target/dependency</Embed-Directory>
+            <Embed-StripGroup>true</Embed-StripGroup>
+            <Import-Package>
+              !sun.misc;resolution:=optional,
+              !com.sun.msv.*;resolution:=optional,
+              !com.sun.jdmk.comm;resolution:=optional,
+              sun.io.*;resolution:=optional,
+              com.beust.jcommander*;resolution:=optional,
+              com.google.appengine.api.*;resolution:=optional,
+              com.google.apphosting.api.*;resolution:=optional,
+              com.google.appengine.repackaged.*;resolution:=optional,
+              org.apache.log.*;resolution:=optional,
+              org.apache.oro.text.perl.*;resolution:=optional,
+              antlr.*;resolution:=optional,
+              org.apache.tools.ant.*;resolution:=optional,
+              junit.framework.*;resolution:=optional,
+              org.dom4j.*;resolution:=optional,
+              org.jdom.*;resolution:=optional,
+              org.bouncycastle.*;resolution:=optional,
+              com.sun.org.apache.xerces.internal.*;resolution:=optional,
+              *</Import-Package>
+            <_noee>true</_noee>
```

Comment:
Why is this needed?

Reply:
If I remove the `_noee` directive, I get the error below and the bundle doesn't activate. I'll take a look at your plugin related questions and circle back to this.

06.04.2020 15:51:23.218 *ERROR* [Background Update org.apache.sling.auth.saml2 (179)] org.apache.felix.http.jetty Cannot install or update bundle from /var/folders/ww/vd006cgx4zgdvvxcfk02b5xr0000gp/T/install2386711195585220288.tmp (org.osgi.framework.BundleException: Unable to resolve org.apache.sling.auth.saml2 [179](R 179.190): missing requirement [org.apache.sling.auth.saml2 [179](R 179.190)] osgi.ee; (&(osgi.ee=JavaSE)(version=9.0)) Unresolved requirements: [[org.apache.sling.auth.saml2 [179](R 179.190)] osgi.ee; (&(osgi.ee=JavaSE)(version=9.0))])
org.osgi.framework.BundleException: Unable to resolve org.apache.sling.auth.saml2 [179](R 179.190): missing requirement [org.apache.sling.auth.saml2 [179](R 179.190)] osgi.ee; (&(osgi.ee=JavaSE)(version=9.0)) Unresolved requirements: [[org.apache.sling.auth.saml2 [179](R 179.190)] osgi.ee; (&(osgi.ee=JavaSE)(version=9.0))]
	at org.apache.felix.framework.Felix.resolveBundleRevision(Felix.java:4133)
	at org.apache.felix.framework.Felix.startBundle(Felix.java:2118)
	at org.apache.felix.framework.BundleImpl.start(BundleImpl.java:998)
	at org.apache.felix.webconsole.internal.core.BaseUpdateInstallHelper.run(BaseUpdateInstallHelper.java:192)
	at java.lang.Thread.run(Thread.java:748)

Reply:
After changing to bnd-maven-plugin the noee is gone and everything seems to work with the new bundle.